src/classes/BDI_CustObjMappingGAUAllocation.cls

```diff
@@ -0,0 +1,103 @@
+
+        Map<String, Schema.DescribeFieldResult> sourceFieldDescribeMap = UTIL_Describe.getAllFieldsDescribe(UTIL_Namespace.adjustNPSPNamespace('npsp__DataImport__c'));
+
+        String PAYMENT_FIELDNAME = UTIL_Namespace.adjustNPSPNamespace('npsp__Payment__c');
+        String OPPORTUNITY_FIELDNAME = UTIL_Namespace.adjustNPSPNamespace('npsp__Opportunity__c');
+        String CAMPAIGN_FIELDNAME = UTIL_Namespace.adjustNPSPNamespace('npsp__Campaign__c');
+
+        for (BDI_ObjectWrapper objWrap : objWraps) {
+            Map<String, Schema.DescribeFieldResult> fieldDescribeMap = UTIL_Describe.getAllFieldsDescribe(objWrap.objMapping.Object_API_Name__c);
+
+            objWrap.sObj = Schema.getGlobalDescribe().get(objWrap.objMapping.Object_API_Name__c).newSObject();
+
+            if (objWrap.existingSObjectId != null) {
+                objWrap.sObj.put('Id', objWrap.existingSObjectId);
+            }
+
+            Object campaignValue;
+            Object paymentValue;
+            Object opportunityValue;
+
+            for (Data_Import_Field_Mapping__mdt fieldMapping : objWrap.fieldMappings) {
+                String sourceFieldName = fieldMapping.Source_Field_API_Name__c;
+                String targetFieldName = fieldMapping.Target_Field_API_Name__c;
+
+                Object value = objWrap.dataImport.get(sourceFieldName);
+
+                Schema.DescribeFieldResult sourceFieldDescribe = sourceFieldDescribeMap.get(sourceFieldName.toLowerCase());
+                Schema.DescribeFieldResult targetFieldDescribe = fieldDescribeMap.get(targetFieldName.toLowerCase());
+
+                //Confirm that it is real field, and that it is accessible to the running user.
+                if (targetFieldDescribe != null && targetFieldDescribe.isAccessible()) {
+                    //Confirm that either the sObj Id is null (ie new record) or that the field is updatable if it is not new.
+                    if (objWrap.sObj.Id == null || targetFieldDescribe.isUpdateable()) {
+                        // If the target field mapping is one of the key parent objects, then extract the values for later logic
+                        if (fieldMapping.Target_Field_API_Name__c == PAYMENT_FIELDNAME) {
+                            paymentValue = value;
+                        } else if (fieldMapping.Target_Field_API_Name__c == OPPORTUNITY_FIELDNAME) {
+                            opportunityValue = value;
+                        } else if (fieldMapping.Target_Field_API_Name__c == CAMPAIGN_FIELDNAME) {
+                            campaignValue = value;
+                        } else {
+                            castAndCopyField(objWrap, sourceFieldName, sourceFieldDescribe, targetFieldName, targetFieldDescribe);
+                        }
+                    }
+                }
+            }
+
+            // If the payment is specified and payment allocations are enabled, always use that first.
```

Comment:
If the payment allocations are not enabled then it will use opportunity instead regardless of payment value. I've verified this with Alison as the desired behavior.

Reply:
👍🏽